### ISSUE TITLE
Silicon could support bsg_time()

### ISSUE THREAD

**Issue by vb000** [cornell-brg/hb-pytorch#130]
Update silicon target to use bsg_time() when it's available. It can probably supported using the fpga on the board.

**Comment by drichmond** [comment]
What specifically do you need from bsg_time? I deprecated that in favor of the more general API that reads a cycle counter from the manycore hardware

**Comment by drichmond** [comment]
That API was implemented in this PR https://github.com/bespoke-silicon-group/bsg_replicant/pull/601

**Comment by vb000** [comment]
Either `bsg_time` or `hb_mc_manycore_get_cycle`... can this supported using on board fpga?

**Comment by drichmond** [comment]
`bsg_time` --> no
`*_get_cycle` -> yes

Does the FPGA board (for the silicon) use Leonard's interface? I just added it as a MMIO register

Does the FPGA have access to the Manycore clock?

**Comment by vb000** [comment]
I'd have to check with Paul...

(But Manycore's clock is fast and fpga might not be timing closed with manycore's clock?)

**Comment by drichmond** [comment]
Yeah. If we had a proxy for the manycore clock though, that would be useful (like, clk/10). 

Otherwise you can just use a cycle counter on the FPGA a multiply by N to get the manycore cycles (where N is the ratio between core clock and FPGA clock)

**Comment by drichmond** [comment]
This is the implementation: https://github.com/bespoke-silicon-group/bsg_replicant/blob/ff4704ebe789512cdbcb10094efd9be6f78e82a5/libraries/platforms/aws-fpga/hardware/cl_manycore.sv#L1148

**Comment by vb000** [comment]
That make me realize, why can't we just use system time? Silicon time is real right?

**Comment by taylor-bsg** [comment]
Regarding what we need, did Cornell want cycle count or just a measure of
absolute time for their profiling?  We can get the latter in the FPGA off
of the FPGA clock. The core clock does run out of the chip, but it's too
fast for the FPGA to look at, and it also varies with chip execution.

M

**Comment by drichmond** [comment]
Cornell explicitly asked for cycle count (I can send relevant slack messages).

`bsg_time` is deprecated and is removed in a PR that is under review. It was a hack, and it was only intended for printing debug messages. In simulation, it would print what `$time` reports, and switched to wallclock time for execution. I added it back when we were writing the runtime libraries so that we would know where to look in the waveforms when the simulations failed.

**Comment by taylor-bsg** [comment]
Sure, but are they using cycle count to compute kernel execution time?

**Comment by yodada** [comment]
Yep. Our hope is to get cycles and calculate execution time by assuming a frequency
